cmd/kubermatic-installer/cmd_mirror_images.go

```diff
@@ -115,197 +126,225 @@ func MirrorImagesFunc(logger *logrus.Logger, versions kubermaticversion.Versions
+			}
 
-		kubermaticConfig, err := defaulting.DefaultConfiguration(config, zap.NewNop().Sugar())
-		if err != nil {
-			return fmt.Errorf("failed to default KubermaticConfiguration: %w", err)
-		}
+			// if we pass the option to ignore repository overrides in the KubermaticConfiguration,
+			// we make sure we omit any repository configured in the loaded config so they get
+			// properly defaulted.
+			if options.IgnoreRepositoryOverrides {
+				config.Spec.API.DockerRepository = ""
+				config.Spec.UI.DockerRepository = ""
+				config.Spec.MasterController.DockerRepository = ""
+				config.Spec.SeedController.DockerRepository = ""
+				config.Spec.Webhook.DockerRepository = ""
+				config.Spec.UserCluster.KubermaticDockerRepository = ""
+				config.Spec.UserCluster.DNATControllerDockerRepository = ""
+				config.Spec.UserCluster.EtcdLauncherDockerRepository = ""
+				config.Spec.UserCluster.Addons.DockerRepository = ""
+				config.Spec.VerticalPodAutoscaler.Recommender.DockerRepository = ""
+				config.Spec.VerticalPodAutoscaler.Updater.DockerRepository = ""
+				config.Spec.VerticalPodAutoscaler.AdmissionController.DockerRepository = ""
+			}
 
-		clusterVersions, err := images.GetVersions(logger, kubermaticConfig, options.VersionFilter)
-		if err != nil {
-			return fmt.Errorf("failed to load versions: %w", err)
-		}
+			kubermaticConfig, err := defaulting.DefaultConfiguration(config, zap.NewNop().Sugar())
+			if err != nil {
+				return fmt.Errorf("failed to default KubermaticConfiguration: %w", err)
+			}
 
-		ctx := cmd.Context()
+			clusterVersions, err := images.GetVersions(logger, kubermaticConfig, options.VersionFilter)
+			if err != nil {
+				return fmt.Errorf("failed to load versions: %w", err)
+			}
+
+			// if no local addons path is given, use the configured addons
+			// Docker image and extract the addons from there
+			if options.AddonsPath == "" {
+				addonsImage := options.AddonsImage
+				if addonsImage == "" {
+					suffix := kubermaticConfig.Spec.UserCluster.Addons.DockerTagSuffix
+
+					tag := versions.Kubermatic
+					if suffix != "" {
+						tag = fmt.Sprintf("%s-%s", versions.Kubermatic, suffix)
+					}
 
-		// if no local addons path is given, use the configured addons
-		// Docker image and extract the addons from there
-		if options.AddonsPath == "" {
-			addonsImage := options.AddonsImage
-			if addonsImage == "" {
-				suffix := kubermaticConfig.Spec.UserCluster.Addons.DockerTagSuffix
+					addonsImage = kubermaticConfig.Spec.UserCluster.Addons.DockerRepository + ":" + tag
+				}
+
+				if addonsImage != "" {
+					tempDir, err := images.ExtractAddons(ctx, logger, addonsImage)
+					if err != nil {
+						return fmt.Errorf("failed to create local addons path: %w", err)
+					}
+					defer os.RemoveAll(tempDir)
 
-				tag := versions.Kubermatic
-				if suffix != "" {
-					tag = fmt.Sprintf("%s-%s", versions.Kubermatic, suffix)
+					options.AddonsPath = tempDir
 				}
+			}
```

Comment:
since the linter is complaining about `MirrorImagesFunc` being too complex - can you please move this whole figuring out the configuration and validating flags into its own function?